src/NuGet.Clients/NuGet.SolutionRestoreManager/BrokeredServicesUtility.cs

```diff
@@ -11,11 +11,19 @@ namespace NuGet.SolutionRestoreManager
     internal static class BrokeredServicesUtility
     {
         // It is very important that these names and versions are kept the same as the ones in NuGet.VisualStudio.Internal.Contracts.NuGetServices.
-        internal const string NuGetSolutionServiceName = "NuGetSolutionService";
-        internal const string NuGetSolutionServiceVersion = "1.0.0";
+        internal const string SolutionServiceName = "Microsoft.VisualStudio.NuGet.SolutionService";
+        internal const string SolutionServiceVersion = "1.0.0";
 
-        internal static readonly ServiceRpcDescriptor NuGetSolutionService = new ServiceJsonRpcDescriptor(
-            new ServiceMoniker(NuGetSolutionServiceName, new Version(NuGetSolutionServiceVersion)),
+        internal const string DeprecatedSolutionServiceName = "NuGetSolutionService";
+        internal const string DeprecatedSolutionServiceVersion = "1.0.0";
+
+        internal static readonly ServiceRpcDescriptor DeprecatedSolutionService = new ServiceJsonRpcDescriptor(
```

Comment:
probably unused.

Reply:
I will check to see if there is telemetry for brokered services usage but in general we shipped it so we should probably support until we can determine the impact.

Reply:
Nah it's the RPC descriptor that's unused. 

as far as removign the service completely goes, the current service is only exposed in Codespaces scenarios, so I wouldn't expect customers to depend on it publicly yet.

Reply:
The scenario would be that you have an old client trying to connect to a codespace server that has been updated, it would fail at that point.

Reply:
I understand now, you are talking about removing the descriptor, yea I agree we should not be exposing that for deprecated services anyways.